npm/namespace.go

```diff
@@ -135,6 +132,14 @@ func (npMgr *NetworkPolicyManager) AddNamespace(nsObj *corev1.Namespace) error {
 	return nil
 }
 
+// AddNamespaceWithLock acquires NetworkPolicyManager lock before adding namespace to ipset
```

Comment:
I think a cleaner way to do this is to unlock & lock in the caller instead of having this function.

Reply:
https://github.com/Azure/azure-container-networking/pull/461/files/63dd9d5784f75c11d09d573d5274d90a773bd646#diff-88c55b4ac2573a453141838261b6b413R91

AddNamespace used to take a bool to determine that, but having every signature with a bool seemed a bit like an antipattern, but it can be changed back

Reply:
That was the patter I had it in before. And I don't feel strongly about either to change it back and forth